### ISSUE TITLE
[Bug] CuteDSL paged decode with `is_causal=False` uses padding KV lanes

### ISSUE THREAD

**Issue by elwhyjay** [flashinfer-ai/flashinfer#3716]
## Environment

- GPU: NVIDIA B200, `sm_100`
- FlashInfer: `0.6.13rc2` and current `main` (`015f5973`)
- PyTorch: `2.9.1+cu128`
- CUDA driver: 580.126.09

## Description

`BatchDecodePagedCuteDSLWrapper` gives wrong output when `is_causal=False` and `seq_len` is not aligned to the 256-token sequence tile.
The non-causal path skips the boundary mask pass. KV lanes after `seq_len` remain in the score tile and enter the softmax denominator. With `q = 0`, attention should be uniform over valid KV tokens only, but the output is scaled as if padding lanes were valid.
The causal path is correct.

## Minimal Repro

```python
import torch
from flashinfer.cute_dsl.attention import BatchDecodePagedCuteDSLWrapper

D, page, H = 128, 16, 8
dev = torch.device("cuda")
dt = torch.bfloat16


def check(seq_len, is_causal):
    npages = (seq_len + page - 1) // page
    indptr = torch.tensor([0, npages], dtype=torch.int32, device=dev)
    indices = torch.arange(npages, dtype=torch.int32, device=dev)
    seq_lens = torch.tensor([seq_len], dtype=torch.int32, device=dev)

    torch.manual_seed(1)
    k_cache = torch.randn(npages, page, H, D, device=dev, dtype=dt) * 0.1
    v_cache = torch.randn(npages, page, H, D, device=dev, dtype=dt) * 0.1
    q = torch.zeros(1, H, D, device=dev, dtype=dt)
    ref = v_cache.reshape(-1, H, D)[:seq_len].float().mean(0)

    wrapper = BatchDecodePagedCuteDSLWrapper(
        torch.empty(512 * 1024 * 1024, dtype=torch.uint8, device=dev)
    )
    wrapper.plan(
        indptr,
        indices,
        seq_lens,
        num_qo_heads=H,
        num_kv_heads=H,
        head_dim=D,
        page_size=page,
        q_data_type=dt,
        kv_data_type=dt,
        sm_scale=1.0 / (D**0.5),
        reduction="none",
        kv_splits=1,
        q_len_per_req=1,
        is_causal=is_causal,
        max_kv_len=seq_len,
    )
    out = wrapper.run(q, k_cache, v_cache).reshape(H, D).float()
    torch.cuda.synchronize()
    ratio = (ref.abs().mean() / (out.abs().mean() + 1e-12)).item()
    max_abs = (out - ref).abs().max().item()
    print(is_causal, seq_len, ratio, max_abs)


for s in (16, 64, 128, 240, 256, 257, 512):
    check(s, is_causal=False)
for s in (16, 64, 128, 240, 256, 257, 512):
    check(s, is_causal=True)
```

## Observed

Before the fix:

```text
is_causal=False seq_len=  16  ref/out=16.000  max_abs=8.61e-02
is_causal=False seq_len=  64  ref/out= 4.000  max_abs=3.89e-02
is_causal=False seq_len= 128  ref/out= 2.000  max_abs=1.63e-02
is_causal=False seq_len= 240  ref/out= 1.067  max_abs=1.33e-03
is_causal=False seq_len= 256  ref/out= 1.000  max_abs=6.02e-05
is_causal=False seq_len= 257  ref/out= 1.940  max_abs=1.09e-02
is_causal=True  all tested seq_len values: ref/out=1.000
```

## Expected

`is_causal=False` should mask KV lanes where `key_pos >= seq_len`. All tested sequence lengths should have `ref/out` close to 1.0.
